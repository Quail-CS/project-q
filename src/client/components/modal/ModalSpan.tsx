/**
 * @module ModalSpan
 * @description Modal Span React Component
 */

import * as React from 'react';

interface IModalSpanProps {
  children: string;
  header: boolean
};

const ModalSpan = (props: IModalSpanProps): JSX.Element => {
  const { children, header } = props;
  const spanClass = header ? 'modal-header-text' : 'modal-body-text';
  return (
    <span className={spanClass}>
      {children}
    </span>
  );
};

export default ModalSpan;
