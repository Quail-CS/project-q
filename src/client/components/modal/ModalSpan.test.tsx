/**
 * @module ModalText.test
 * @description Unit Tests for Modal Text React Component
 */

import * as Adapter from 'enzyme-adapter-react-16';
import * as Enzyme from 'enzyme';
import * as React from 'react';
import * as Renderer from 'react-test-renderer';

import ModalSpan from './ModalSpan';

Enzyme.configure({ adapter: new Adapter() });

// Declare constants for testing
const stubModalHeaderText = 'Reset Ethiq?';
const stubModalBodyText = 'Continue?';

describe('Modal Text Component: ', () => {
  it('matches the snapshot', () => {
    const tree = Renderer.create(<ModalSpan header={true}>{stubModalHeaderText}</ModalSpan>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when header is true, renders a span with class "modal-header-text', () => {
    const shallowModalSpanWrapper = Enzyme.shallow(<ModalSpan header={true}>{stubModalHeaderText}</ModalSpan>);
    expect(shallowModalSpanWrapper.hasClass('modal-header-text')).toEqual(true);
    expect(shallowModalSpanWrapper.type()).toEqual('span');
  });

  it('when header is false, renders a span with class "modal-body-text', () => {
    const shallowModalSpanWrapper = Enzyme.shallow(<ModalSpan header={false}>{stubModalBodyText}</ModalSpan>);
    expect(shallowModalSpanWrapper.hasClass('modal-body-text')).toEqual(true);
    expect(shallowModalSpanWrapper.type()).toEqual('span');
  });

  it('renders the text passed in as children in the span', () => {
    const shallowModalHeaderSpanWrapper = Enzyme.shallow(<ModalSpan header={true}>{stubModalHeaderText}</ModalSpan>);
    expect(shallowModalHeaderSpanWrapper.text()).toEqual(stubModalHeaderText);
    const shallowModalBodySpanWrapper = Enzyme.shallow(<ModalSpan header={false}>{stubModalBodyText}</ModalSpan>);
    expect(shallowModalBodySpanWrapper.text()).toEqual(stubModalBodyText);
  });
});
