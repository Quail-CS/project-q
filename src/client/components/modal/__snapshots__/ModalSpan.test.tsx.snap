// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Text Component:  matches the snapshot 1`] = `
<span
  className="modal-header-text"
>
  Reset Ethiq?
</span>
`;
